system_tests/das_test.go

```diff
@@ -132,12 +172,17 @@ func TestDASRekey(t *testing.T) {
 		l2clientA := ClientForStack(t, l2stackA)
 
 		l1NodeConfigB.BlockValidator.Enable = false
-		l1NodeConfigB.DataAvailability.Enable = true
-		l1NodeConfigB.DataAvailability.AggregatorConfig = aggConfigForBackend(t, backendConfigA)
+		lifecycleManager, restServer, restFulDataAvailabilityConfig := startLocalRestfulDasServer(t, ctx, dasDataDir)
+		l1NodeConfigB.DataAvailability = restFulDataAvailabilityConfig
 		l2clientB, _, l2stackB := Create2ndNodeWithConfig(t, ctx, nodeA, l1stack, &l2info.ArbInitData, l1NodeConfigB)
 		checkBatchPosting(t, ctx, l1client, l2clientA, l1info, l2info, big.NewInt(1e12), l2clientB)
 		requireClose(t, l2stackA)
 		requireClose(t, l2stackB)
+
+		err = restServer.Shutdown()
+		Require(t, err)
```

Comment:
What's restServer being used for here? It just looks like it's being shut down after being started.

Reply:
So startLocalRestfulDasServer create the rest server for a listener [here](https://github.com/OffchainLabs/nitro/pull/864/files#diff-1b0ce41d13751e044454b13fb3b475b30c60245c97724fdb84db363fa6f96f4bR111), and the listener is passed as config to  restFulDataAvailabilityConfig that config is used in l1NodeConfigB, which is used to create l2clientB.

l2clientB is used by checkBatchPosting to make the eth call.

restServer.shutdown  closes the connection opened by the listener.(This is done in a similar manner [here](https://github.com/OffchainLabs/nitro/blob/master/system_tests/das_test.go#L405))